external-crates/move/crates/move-package/src/lib.rs

```diff
@@ -215,4 +217,27 @@ impl BuildConfig {
             .set_warnings_are_errors(self.warnings_are_errors)
             .set_silence_warnings(self.silence_warnings)
     }
+
+    pub fn update_lock_file_toolchain_version(&self, compiler_version: String) -> Result<()> {
+        let lock_file = self.lock_file.as_ref();
+        if lock_file.is_none() {
+            return Ok(());
+        };
```

Comment:
Currently when we invoke framework build on compile, we don't set a lock file, and this check helps us make progress (we don't want to fail hard). Essentially, the caller of `update_lock_file_toolchain_version` is responsible for ensuring `lock_file` exists, or it won't be updated by this logic.